hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java

```diff
@@ -74,10 +74,24 @@ public void test_itemStream_withWindowing() throws Throwable {
          .withNativeTimestamps(0)
          .window(WindowDefinition.tumbling(1000))
          .aggregate(AggregateOperations.counting())
-         .apply(assertCollectedEventually(10, windowResults -> {
-             // find any window that has 10 items, some may be incomplete due to hiccups
-             boolean matched = windowResults.stream().anyMatch(r -> r.result() == itemsPerSecond);
-             assertTrue("Did not find any window with 10 items: " + windowResults, matched);
+         .apply(assertCollectedEventually(60, windowResults -> {
+             //look at last 5 windows at most
+             int windowsToConsider = Math.min(5, windowResults.size());
```

Comment:
now if just 1 window has 9 items, the test will pass. That's not a very strong assertion.

Reply:
If there is 1 window with 9 items, then `deviationFromTarget = 1`, which is not smaller than the allowed tolerance of `0.01`, so it won't pass, but wait for more windows.

Reply:
ok, again I read it thinking it was a percentage, but it's actually an absolute value.

Reply:
how long does the test take to pass?

Reply:
On my machine it always takes 5 seconds to pass, but unfortunately it's not the test that's bad, but the first window always has way less values than it should, usually at most 7, instead of 10. 

And that usually only in the "member" mode, the subsequent "client" mode is better, usually starts with a 9. 

Feels like a warmup, but I'm not sure what it is. Looking at it.

Reply:
how about we always skip the first window?

Reply:
Good idea, pushed that version, much better.